cdap-client-tests/src/test/java/co/cask/cdap/client/PreferencesClientTestRun.java

```diff
@@ -83,6 +84,8 @@ public void testProgramAPI() throws Exception {
                                                         AppReturnsArgs.SERVICE));
       programClient.start(AppReturnsArgs.NAME, ProgramType.SERVICE, AppReturnsArgs.SERVICE,
                           ImmutableMap.of("run", "value"));
```

Comment:
Seems like we don't. We have `ServiceManager` (unit-test framework) and `AbstractContext` that has a very similar `getURL` method that supports timeout, but not in the `ServiceClient`. Need to revisit the client library again.